srcopsmetrics/github_knowledge.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env python3
+# SrcOpsMetrics
+# Copyright(C) 2020 Francesco Murdaca, Dominik Tuchyna
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""A base class for collecting bot knowledge from GitHub."""
+
+import os
+import logging
+import time
+import json
+from datetime import datetime
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Set
+from typing import Tuple
+from typing import Union
+
+from pathlib import Path
+
+from github import Github
+from github import GithubObject
+from github import Issue
+from github import IssueComment
+from github import PullRequest
+from github import PullRequestReview
+from github import PaginatedList
+from github.Repository import Repository
+
+from srcopsmetrics.enums import EntityTypeEnum
+from srcopsmetrics.github_knowledge_store import GitHubKnowledgeStore
+
+_LOGGER = logging.getLogger(__name__)
+
+_GITHUB_ACCESS_TOKEN = os.getenv("GITHUB_ACCESS_TOKEN")
+
+ISSUE_KEYWORDS = {"close", "closes", "closed", "fix",
+                  "fixes", "fixed", "resolve", "resolves", "resolved"}
+
+STANDALONE_LABELS = {"size"}
+
+
+class GitHubKnowledge:
+    """Context manager for entity extraction process from GitHub."""
```

Comment:
I think that this does not count as a context manager yet (they have exit and enter methods and you can use them in the manner 'with manager(...) as ....:'

this is currently rather a set of methods wrapped in a class
also, some of them are static and some of them are not, is there any specific reason for that?

but i think the idea behind making this class a context manager is great! It would be awesome if we would use it as a context manager

Reply:
Not at all, you are right, I have to change the docstring here actually. Some method are static as some methods are not bounded to the class actually, aka, no dependencies from other functions of the class or parameters of the class.

Reply:
changes addressed here: https://github.com/AICoE/SrcOpsMetrics/pull/54/commits/29d15a1017f548ac284d02a01af9041ba3bac408